src/main/java/com/example/managershop/service/Impl/RoleServiceImpl.java

```diff
@@ -1,11 +1,42 @@
 package com.example.managershop.service.Impl;
 
+import com.example.managershop.dao.RoleRepository;
+import com.example.managershop.entities.Role;
+import com.example.managershop.entities.User;
 import com.example.managershop.service.RoleService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import javax.transaction.Transactional;
 
 @Service
 @Transactional
 public class RoleServiceImpl implements RoleService {
+
+    @Autowired
+    private RoleRepository roleRepository;
+    @Override
+    public Role updateRole(Long idRole, Role newRole) {
+
+        Role role= loadRoleById(idRole);
+        if(role.equals(null)) throw new RuntimeException("role not exist");
+        role.setNameRole(newRole.getNameRole());
+        role.setUsers(role.getUsers());
+        role.setLevelPriorite(role.getLevelPriorite());
+        roleRepository.save(role);
+        return  role;
+    }
+
+    @Override
+    public String deleteRole(Role role) {
+        if(loadRoleById(role.getIdRole()).equals(null)) throw new RuntimeException("Role not exist");
```

Comment:
meme commentaire sur la classe Client

Reply:
if( ObjectUtils.equals(loadRoleById(role.getIdRole()),null)) throw new RoleNotFoundException("Role not exist");  
// RoleNotFoundException est le role que j'ai personnalisé